File: CodeWars/7kyu/JavaScript/018-get-the-middle-character.js
// https://www.codewars.com/kata/56747fd5cb988479af000028
// You are going to be given a word. Your job is to return the middle character of the word. If the word's length is odd, 
// return the middle character. If the word's length is even, return the middle 2 characters.
// Kata.getMiddle("test") should return "es"
// Kata.getMiddle("testing") should return "t"
// Kata.getMiddle("middle") should return "dd"
// Kata.getMiddle("A") should return "A"

function getMiddle(s) {
    let middle = Math.floor(s.length / 2);
    let even = s.length % 2 === 0;

    return (even ? s.charAt(middle - 1) : '') + s.charAt(middle)
}


console.log(getMiddle("test"), "es");
console.log(getMiddle("testing"), "t");
console.log(getMiddle("middle"), "dd");
console.log(getMiddle("A"), "A");

File: CodeWars/7kyu/Python/018-get-the-middle-character.py
# https://www.codewars.com/kata/56747fd5cb988479af000028
# You are going to be given a word. Your job is to return the middle character of the word.
# If the word's length is odd, return the middle character. If the word's length is even, return the middle 2 characters.
# Kata.getMiddle("test") should return "es"
# Kata.getMiddle("testing") should return "t"

def get_middle(s):
    if len(s) % 2 == 0:
        start = len(s) // 2 - 1
        end = start + 2
    else:
        start = len(s) // 2
        end = start + 1

    return s[start:end]


print(get_middle("test"), "es")
print(get_middle("testing"), "t")
print(get_middle("middle"), "dd")
print(get_middle("A"), "A")
print(get_middle("of"), "of")
